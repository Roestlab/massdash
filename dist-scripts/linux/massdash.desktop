[Desktop Entry]
Version=0.0.8
Type=Application
Name=massdash
Exec=/usr/local/bin/massdash
Icon=/usr/share/icons/massdash.ico
Terminal=true
