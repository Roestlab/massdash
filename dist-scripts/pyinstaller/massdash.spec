# -*- mode: python ; coding: utf-8 -*-

import site
import importlib.metadata
import os
import sys

from PyInstaller.building.build_main import Analysis, PYZ, EXE, COLLECT
from PyInstaller.utils.hooks import collect_all, collect_data_files, copy_metadata

from transformers.dependency_versions_check import pkgs_to_check_at_runtime

sys.setrecursionlimit(sys.getrecursionlimit() * 5)

##################### User definitions
exe_name = 'massdash_gui'
script_name = 'run.py'
if sys.platform[:6] == "darwin":
	icon = '../../massdash/assets/img/MassDash_Logo.icns'
else:
	icon = '../../massdash/assets/img/MassDash_Logo.ico'
block_cipher = None
location = os.getcwd()
project = "massdash"
remove_tests = True
bundle_name = "massdash"
#####################


requirements = {project, "streamlit_javascript", "upsetplot", "distributed"}
datas = [(f"{site.getsitepackages()[0]}/streamlit/runtime", "./streamlit/runtime")]
hidden_imports = set()
binaries = []
checked = set()
while requirements:
	requirement = requirements.pop()
	print(f"Info: Checking {requirement}")
	checked.add(requirement)
	if requirement in ["pywin32"]:
		continue
	try:
		module_version = importlib.metadata.version(requirement)
	except (
		importlib.metadata.PackageNotFoundError,
		ModuleNotFoundError,
		ImportError
	):
		continue
	try:
		datas_, binaries_, hidden_imports_ = collect_all(
			requirement,
			include_py_files=True
		)
	except ImportError:
		continue
	datas += datas_
	# binaries += binaries_
	hidden_imports_ = set(hidden_imports_)
	if "" in hidden_imports_:
		hidden_imports_.remove("")
	if None in hidden_imports_:
		hidden_imports_.remove(None)
	requirements |= hidden_imports_ - checked
	hidden_imports |= hidden_imports_

if remove_tests:
	hidden_imports = sorted(
		[h for h in hidden_imports if "test" not in h.split(".")]
	)
else:
	hidden_imports = sorted(hidden_imports)


hidden_imports = [h for h in hidden_imports if "__pycache__" not in h]
datas = [d for d in datas if ("__pycache__" not in d[0]) and (d[1] not in [".", "Resources", "scripts"])]

#if sys.platform[:5] == "win32":
#	base_path = os.path.dirname(sys.executable)
#	library_path = os.path.join(base_path, "Library", "bin")
#	dll_path = os.path.join(base_path, "DLLs")
#	libcrypto_dll_path = os.path.join(dll_path, "libcrypto-1_1-x64.dll")
#	libssl_dll_path = os.path.join(dll_path, "libssl-1_1-x64.dll")
#	libcrypto_lib_path = os.path.join(library_path, "libcrypto-1_1-x64.dll")
#	libssl_lib_path = os.path.join(library_path, "libssl-1_1-x64.dll")
#	if not os.path.exists(libcrypto_dll_path):
#		datas.append((libcrypto_lib_path, "."))
#	if not os.path.exists(libssl_dll_path):
#		datas.append((libssl_lib_path, "."))

for _pkg in ["python","accelerate"]:
	if _pkg in pkgs_to_check_at_runtime:
		pkgs_to_check_at_runtime.remove(_pkg)
for _pkg in pkgs_to_check_at_runtime:
	datas += copy_metadata(_pkg)


datas += collect_data_files("streamlit")
datas += copy_metadata("streamlit")
datas += collect_data_files("pyopenms")
datas += copy_metadata("pyopenms")
datas += collect_data_files("massdash")
datas += copy_metadata("massdash")

hidden_imports = ['pyopenms', 'massdash', 'torchaudio.lib.libtorchaudio']

a = Analysis(
	[script_name],
	pathex=[location],
	binaries=binaries,
	datas=datas,
	hiddenimports=hidden_imports,
	hookspath=['./hooks'],
	runtime_hooks=[],
	excludes=[],
	win_no_prefer_redirects=False,
	win_private_assemblies=False,
	cipher=block_cipher,
	noarchive=False
)
pyz = PYZ(
	a.pure,
	a.zipped_data,
	cipher=block_cipher
)

if sys.platform[:5] == "linux":
	exe = EXE(
		pyz,
		a.scripts,
		a.binaries,
		a.zipfiles,
		a.datas,
		name=bundle_name,
		debug=False,
		bootloader_ignore_signals=False,
		strip=False,
		upx=True,
		console=True,
		upx_exclude=[],
		icon=icon
	)
else:
	exe = EXE(
		pyz,
		a.scripts,
		# a.binaries,
		a.zipfiles,
		# a.datas,
		exclude_binaries=True,
		name=exe_name,
		debug=False,
		bootloader_ignore_signals=False,
		strip=False,
		upx=True,
		console=True,
		icon=icon
	)
	coll = COLLECT(
		exe,
		a.binaries,
		# a.zipfiles,
		a.datas,
		strip=False,
		upx=True,
		upx_exclude=[],
		name=exe_name
	)