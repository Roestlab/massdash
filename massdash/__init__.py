"""
massdash
~~~~~~~~
TODO: ADD general documentation about how to use the auto API here

Package Structure
-----------------
testing - a test package
"""

__all__ = [ 'structs', 'preprocess', 'loaders', 'ui', 'server', 'testing' ]
__version__ = "0.0.9"
