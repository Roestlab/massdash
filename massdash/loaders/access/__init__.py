"""
massdash/loaders/access
~~~~~~~~~~~~~~~~~~~~~~~

The :mod:`massdash.loaders.access` subpackage contains the low level access to DIA data files. 
"""

from .GenericResultsAccess import GenericResultsAccess
from .MzMLDataAccess import MzMLDataAccess
from .OSWDataAccess import OSWDataAccess
from .ResultsTSVDataAccess import ResultsTSVDataAccess
from .SqMassDataAccess import SqMassDataAccess
from .TransitionPQPDataAccess import TransitionPQPDataAccess
from .TransitionTSVDataAccess import TransitionTSVDataAccess

__all__ = [ "GenericResultsAccess",
            "MzMLDataAccess",
            "OSWDataAccess",
            "ResultsTSVDataAccess",
            "SqMassDataAccess",
            "TransitionPQPDataAccess",
            "TransitionTSVDataAccess"]