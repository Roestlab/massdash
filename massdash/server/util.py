"""
massdash/server/util
~~~~~~~~~~~~~~~~~~~~
"""

import mmap

def get_string_mslevels_from_bool(mslevel_bool_dict):
    """
    Converts a dictionary of boolean values for mslevels to a string.

    Args:
    mslevel_bool_dict : dict
        A dictionary containing boolean values for mslevels.

    Returns:
    mslevel_str : str
        A string representing the selected mslevel.
    """
    if mslevel_bool_dict['ms1']:
        mslevel_str = 'ms1'
    elif mslevel_bool_dict['ms2']:
        mslevel_str = 'ms2'
    elif mslevel_bool_dict['ms1'] and mslevel_bool_dict['ms2']:
        mslevel_str = 'ms1ms2'
    else:
        raise ValueError('No mslevel selected')
    return mslevel_str

def check_ion_mobility(mzml_file, num_lines_to_check=10_000_000):
    """
    Check if the mzML file contains ion mobility data

    Args:
    mzml_file: (str) mzML file to load
    num_lines_to_check: (int) Number of lines to check for "Ion Mobility"

    Returns:
    Return a boolean indicating if the mzML file contains ion mobility data
    """
    has_ion_mobility = False
    with open(mzml_file, 'rb', 0) as file:
        s = mmap.mmap(file.fileno(), 0, access=mmap.ACCESS_READ) 
        for line in iter(s.readline, b""):
            if b'Ion Mobility' in line:
                has_ion_mobility = True
                break 
            if s.tell() > num_lines_to_check:
                break
    return has_ion_mobility