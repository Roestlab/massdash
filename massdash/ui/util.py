"""
massdash/ui/util
~~~~~~~~~~~~~~~~
"""

from os import getcwd
from os.path import dirname
import streamlit as st
from tkinter import Tk, filedialog

def clicked(button):
    """
    Updates the session state to indicate that a button has been clicked.

    Args:
        button (str): The name of the button that was clicked.

    Returns:
        None
    """
    st.session_state.clicked[button] = True
    
def st_mutable_write(text=None):
    """
    Creates a mutable container that can be updated with text.
    
    Args:
        text (str): The text to be written to the container.
        
    Returns:
        st.empty(): The mutable container.
    """
    write_container = st.empty()
    if text is not None:
        write_container.write(text)
    return write_container

def get_parent_directory(input_file: str=None):
    """
    Get the parent directory of a file, if no file is provided, the current working directory is returned.

    Args:
        input_file (str): The path to the feature file.

    Returns:
        str: The parent directory.
    """
    if input_file is None:
        return getcwd()
    return dirname(input_file)

def tk_file_dialog(file_type: list = [], title: str = "Select File", parent_dir: str = getcwd()):
    """
    Creates a Tkinter file dialog for selecting a file.

    Args:
        file_type (list): List of tuples specifying file types and their corresponding extensions.
                           Example: [("Text files", "*.txt"), ("CSV files", "*.csv")]
        title (str): The title of the file dialog.
        parent_dir (str): The path to the parent directory of the file dialog.

    Returns:
        str: The path to the selected file.
    """
    root = Tk()
    root.withdraw()
    file_type.extend([("All files", "*.*")])
    file_path = filedialog.askopenfilename(filetypes=file_type, title=title, initialdir=parent_dir)
    root.destroy()
    return file_path


def display_input_section(title, key_base: str, file_extension: str, dialog_title: str, placeholder: str, st_cols = None):
    """
    Display an input section with a title, file browsing button, and text input field.

    Parameters:
        title (str): The title of the input section.
        key_base (str): The base key used for storing session state variables
        file_extension (str): The file extension to filter when browsing for files.
        dialog_title (str): The title of the file browsing dialog.
        placeholder (str): The placeholder text for the text input field.
        st_cols (streamlit.columns): The columns to display the input section in.

    Returns:
        str: The value entered in the text input field.
    """
    st.subheader(title)
    if st_cols is None:
        st_cols = st.columns([0.05, 0.95], gap="small")
    with st_cols[0]:
        st.write("\n")
        st.write("\n")
        dialog_button = st.button("üìÅ", key=f'{key_base}_browse', help=f"Browse for the {title} file.")
        if dialog_button:
            st.session_state.tmp_input_dict[key_base] = tk_file_dialog(file_extension, dialog_title, get_parent_directory(st.session_state.tmp_input_dict[key_base]))    
    with st_cols[1]:
        input_value = st.text_input("Enter file path", value=st.session_state.tmp_input_dict[key_base],
                                    placeholder=placeholder, key=f'{key_base}_tmp',
                                    help=f"Path to the {title} file ({file_extension})")
    return input_value
