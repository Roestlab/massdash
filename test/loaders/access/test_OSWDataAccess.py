"""
test/loaders/access/test_OSWDataAccess
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
"""

import pytest
from pathlib import Path
import pandas as pd

from massdash.testing import PandasSnapshotExtension
from massdash.loaders.access.OSWDataAccess import OSWDataAccess
from massdash.util import find_git_directory

TEST_PATH = find_git_directory(Path(__file__).resolve()).parent / 'test'

@pytest.fixture
def snapshot_pandas(snapshot):
    return snapshot.use_extension(PandasSnapshotExtension)

@pytest.fixture
def osw_data_access():
    db_path = f"{str(TEST_PATH)}/test_data/osw/ionMobilityTest.osw"
    osw_data_access = OSWDataAccess(db_path)
    yield osw_data_access
    osw_data_access.conn.close()

@pytest.fixture
def osw_data_access2():
    db_path = f"{str(TEST_PATH)}/test_data/osw/ionMobilityTest.osw"
    osw_data_access = OSWDataAccess(db_path)
    yield osw_data_access
    osw_data_access.conn.close()


@pytest.fixture(params=['ionMobilityTest2', None])
def run(request):
    return request.param

def test_getProteinTable(osw_data_access, snapshot_pandas):
    protein_table = osw_data_access.getProteinTable()
    assert snapshot_pandas == protein_table 

def test_getPeptideTable(osw_data_access, snapshot_pandas):
    peptide_table = osw_data_access.getPeptideTable()
    assert isinstance(peptide_table, pd.DataFrame)
    assert snapshot_pandas == peptide_table

def test_getPeptideTableFromProteinID(osw_data_access, snapshot_pandas):
    protein_id = 539
    peptide_table = osw_data_access.getPeptideTableFromProteinID(protein_id)
    assert snapshot_pandas == peptide_table

def test_getPrecursorCharges(osw_data_access, snapshot_pandas):
    fullpeptidename = "ANS(UniMod:21)SPTTNIDHLK(UniMod:259)"
    precursor_charges = osw_data_access.getPrecursorCharges(fullpeptidename)
    assert snapshot_pandas == precursor_charges

def test_getPeptideTransitionInfo(osw_data_access, snapshot_pandas):
    fullpeptidename = "ANS(UniMod:21)SPTTNIDHLK(UniMod:259)"
    charge = 2
    peptide_transition_info = osw_data_access.getPeptideTransitionInfo(fullpeptidename, charge)
    assert snapshot_pandas == peptide_transition_info 

@pytest.mark.parametrize("fullpeptidename,charge", [("AFVDFLSDEIK", 2), ("INVALID", 0)])
def test_getTransitionIDAnnotationFromSequence(osw_data_access, snapshot_pandas, fullpeptidename, charge):
    transition_group_feature = osw_data_access.getTransitionIDAnnotationFromSequence(fullpeptidename, charge)
    assert snapshot_pandas == transition_group_feature 

@pytest.mark.parametrize("fullpeptidename,charge", [("AFVDFLSDEIK", 2), ("INVALID", 0)])
def test_getPrecursorIDFromPeptideAndCharge(osw_data_access, snapshot, fullpeptidename, charge):
    precursor_id = osw_data_access.getPrecursorIDFromPeptideAndCharge(fullpeptidename, charge)
    assert snapshot == precursor_id 

@pytest.mark.parametrize("fullpeptidename,charge,run", [("AFVDFLSDEIK", 2, 'ionMobilityTest2'), ("INVALID", 0, 'ionMobilityTest2'), ("AFVDFLSDEIK", 2, 'INVALID'),])
def test_getTransitionGroupFeatures(osw_data_access, snapshot, fullpeptidename, charge, run):
    transition_group_feature = osw_data_access.getTransitionGroupFeatures(run, fullpeptidename, charge)
    print(transition_group_feature)
    assert snapshot == transition_group_feature

@pytest.mark.parametrize("fullpeptidename,charge,run", [("AFVDFLSDEIK", 2, 'ionMobilityTest2'), ("INVALID", 0, 'ionMobilityTest2'), ("AFVDFLSDEIK", 2, 'INVALID')])
def test_getTransitionGroupFeaturesDf(osw_data_access, snapshot_pandas, run, fullpeptidename, charge):
    transition_group_feature = osw_data_access.getTransitionGroupFeaturesDf(run, fullpeptidename, charge)
    assert snapshot_pandas == transition_group_feature

def test_getRunNames(osw_data_access, snapshot):
    runnames = osw_data_access.getRunNames()
    assert snapshot == runnames 

def test_getIdentifiedPrecursors(osw_data_access, snapshot):
    precursors = osw_data_access.getIdentifiedPrecursors()
    assert precursors == snapshot

def test_getIdentifiedPrecursorIntensities(osw_data_access, run, snapshot_pandas):
    intensities = osw_data_access.getIdentifiedPrecursorIntensities(run=run)
    assert snapshot_pandas == intensities

def test_getIdentifiedProteins(osw_data_access, run, snapshot):
    proteins = osw_data_access.getIdentifiedProteins(run=run)
    assert proteins == snapshot

def test_getIdentifiedPeptides(osw_data_access, run, snapshot):
    peptides = osw_data_access.getIdentifiedPeptides(run=run)
    assert peptides == snapshot

def test_getNumIdentifiedPrecursors(osw_data_access, run, snapshot):
    num_precursors = osw_data_access.getNumIdentifiedPrecursors(run=run)
    assert num_precursors == snapshot

def test_getNumIdentifiedProteins(osw_data_access, run, snapshot):
    num_proteins = osw_data_access.getNumIdentifiedProteins(run=run)
    assert num_proteins == snapshot

def test_getNumIdentifiedPeptides(osw_data_access, run, snapshot):
    num_peptides = osw_data_access.getNumIdentifiedPeptides(run=run)
    assert num_peptides == snapshot

def test_getSoftware(osw_data_access):
    assert osw_data_access.getSoftware() == "OpenSWATH"

def test_getExperimentSummary(osw_data_access, snapshot_pandas):
    experiment_summary = osw_data_access.getExperimentSummary()
    assert snapshot_pandas == experiment_summary

def test_initializeValidScores(osw_data_access2, snapshot):
    # Open test.osw file because has full .osw output unlike ion mobility
    osw_data_access2._initializeValidScores()

    assert snapshot == osw_data_access2.validScores

@pytest.mark.parametrize("score_table,score,context", [("SCORE_MS2", "SCORE", None),
                                                       ("FEATURE_MS2", "VAR_XCORR_COELUTION", None),
                                                       ("FEATURE_MS1", "VAR_XCORR_COELUTION_COMBINED", None),
                                                       ("SCORE_PROTEIN", "SCORE", "global"),
                                                       ("SCORE_PROTEIN", "SCORE", "run-specific"),
                                                       ("SCORE_PROTEIN", "SCORE", "experiment-wide"),
                                                       ("SCORE_PEPTIDE", "SCORE", "global"),
                                                       ("SCORE_PEPTIDE", "SCORE", "run-specific"),
                                                       ("SCORE_PEPTIDE", "SCORE", "experiment-wide")])
def test_getScoreTable(osw_data_access2, score_table, score, context, snapshot_pandas):
    df = osw_data_access2.getScoreTable(score_table, score, context)
    assert snapshot_pandas == df





def test_getCV():

    pass
