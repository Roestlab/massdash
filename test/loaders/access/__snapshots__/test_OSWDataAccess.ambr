# serializer version: 1
# name: test_getIdentifiedPeptides[None]
  dict({
    '20190816_TIMS05_MA_FlMe_diaPASEF_25pc_50ng_A2_1_25.d_mergeFiles': set({
      'AFVDFLSDEIK',
    }),
    '20190816_TIMS05_MA_FlMe_diaPASEF_25pc_50ng_A2_1_27.d_mergeFiles': set({
      'AFVDFLSDEIK',
    }),
    'ionMobilityTest2': set({
      'AFVDFLSDEIK',
    }),
  })
# ---
# name: test_getIdentifiedPeptides[ionMobilityTest2]
  set({
    'AFVDFLSDEIK',
  })
# ---
# name: test_getIdentifiedPrecursors
  dict({
    '20190816_TIMS05_MA_FlMe_diaPASEF_25pc_50ng_A2_1_25.d_mergeFiles': set({
      'AFVDFLSDEIK2',
    }),
    '20190816_TIMS05_MA_FlMe_diaPASEF_25pc_50ng_A2_1_27.d_mergeFiles': set({
      'AFVDFLSDEIK2',
    }),
    'ionMobilityTest2': set({
      'AFVDFLSDEIK2',
    }),
  })
# ---
# name: test_getIdentifiedProteins[None]
  dict({
    '20190816_TIMS05_MA_FlMe_diaPASEF_25pc_50ng_A2_1_25.d_mergeFiles': set({
      'sp|Q07021|C1QBP_HUMAN',
    }),
    '20190816_TIMS05_MA_FlMe_diaPASEF_25pc_50ng_A2_1_27.d_mergeFiles': set({
      'sp|Q07021|C1QBP_HUMAN',
    }),
    'ionMobilityTest2': set({
      'sp|Q07021|C1QBP_HUMAN',
    }),
  })
# ---
# name: test_getIdentifiedProteins[ionMobilityTest2]
  set({
    'sp|Q07021|C1QBP_HUMAN',
  })
# ---
# name: test_getNumIdentifiedPeptides[None]
  dict({
    '20190816_TIMS05_MA_FlMe_diaPASEF_25pc_50ng_A2_1_25.d_mergeFiles': 1,
    '20190816_TIMS05_MA_FlMe_diaPASEF_25pc_50ng_A2_1_27.d_mergeFiles': 1,
    'ionMobilityTest2': 1,
  })
# ---
# name: test_getNumIdentifiedPeptides[ionMobilityTest2]
  1
# ---
# name: test_getNumIdentifiedPrecursors[None]
  dict({
    '20190816_TIMS05_MA_FlMe_diaPASEF_25pc_50ng_A2_1_25.d_mergeFiles': 1,
    '20190816_TIMS05_MA_FlMe_diaPASEF_25pc_50ng_A2_1_27.d_mergeFiles': 1,
    'ionMobilityTest2': 1,
  })
# ---
# name: test_getNumIdentifiedPrecursors[ionMobilityTest2]
  1
# ---
# name: test_getNumIdentifiedProteins[None]
  dict({
    '20190816_TIMS05_MA_FlMe_diaPASEF_25pc_50ng_A2_1_25.d_mergeFiles': 1,
    '20190816_TIMS05_MA_FlMe_diaPASEF_25pc_50ng_A2_1_27.d_mergeFiles': 1,
    'ionMobilityTest2': 1,
  })
# ---
# name: test_getNumIdentifiedProteins[ionMobilityTest2]
  1
# ---
# name: test_getPrecursorIDFromPeptideAndCharge[AFVDFLSDEIK-2]
  33018
# ---
# name: test_getPrecursorIDFromPeptideAndCharge[INVALID-0]
  None
# ---
# name: test_getRunNames
  list([
    '20190816_TIMS05_MA_FlMe_diaPASEF_25pc_50ng_A2_1_25.d_mergeFiles',
    'ionMobilityTest2',
    '20190816_TIMS05_MA_FlMe_diaPASEF_25pc_50ng_A2_1_27.d_mergeFiles',
  ])
# ---
# name: test_getTransitionGroupFeatures[AFVDFLSDEIK-2-INVALID]
  list([
  ])
# ---
# name: test_getTransitionGroupFeatures[AFVDFLSDEIK-2-ionMobilityTest2]
  list([
    -------- TransitionGroupFeature --------
  leftBoundary: 6235.8486328125
  rightBoundary: 6248.42822265625
  areaIntensity: 2848190.0
  consensusApex: 6242.15
  consensusApexIntensity: 352642.16135025
  qvalue: 3.5084067486223456e-05
  consensusApexIM: 0.978579389257473
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AFVDFLSDEIK
  software: OpenSWATH,
    -------- TransitionGroupFeature --------
  leftBoundary: 6255.64599609375
  rightBoundary: 6266.51513671875
  areaIntensity: 35433.9
  consensusApex: 6256.67
  consensusApexIntensity: 2888.98703575134
  qvalue: 0.0001776217262565
  consensusApexIM: 0.981810896830182
  precursor_mz: None
  precursor_charge: 2
  product_annotations: None
  product_mz: None
  sequence: AFVDFLSDEIK
  software: OpenSWATH,
  ])
# ---
# name: test_getTransitionGroupFeatures[INVALID-0-ionMobilityTest2]
  list([
  ])
# ---
# name: test_initializeValidScores
  dict({
    'FEATURE_MS1': list([
      'VAR_MASSDEV_SCORE',
      'VAR_MI_CONTRAST_SCORE',
      'VAR_MI_COMBINED_SCORE',
      'VAR_ISOTOPE_CORRELATION_SCORE',
      'VAR_ISOTOPE_OVERLAP_SCORE',
      'VAR_XCORR_COELUTION_CONTRAST',
      'VAR_XCORR_COELUTION_COMBINED',
      'VAR_XCORR_SHAPE_CONTRAST',
      'VAR_XCORR_SHAPE_COMBINED',
    ]),
    'FEATURE_MS2': list([
      'VAR_BSERIES_SCORE',
      'VAR_DOTPROD_SCORE',
      'VAR_INTENSITY_SCORE',
      'VAR_ISOTOPE_CORRELATION_SCORE',
      'VAR_ISOTOPE_OVERLAP_SCORE',
      'VAR_LIBRARY_CORR',
      'VAR_LIBRARY_DOTPROD',
      'VAR_LIBRARY_MANHATTAN',
      'VAR_LIBRARY_RMSD',
      'VAR_LIBRARY_ROOTMEANSQUARE',
      'VAR_LIBRARY_SANGLE',
      'VAR_LOG_SN_SCORE',
      'VAR_MANHATTAN_SCORE',
      'VAR_MASSDEV_SCORE',
      'VAR_MASSDEV_SCORE_WEIGHTED',
      'VAR_MI_SCORE',
      'VAR_MI_WEIGHTED_SCORE',
      'VAR_MI_RATIO_SCORE',
      'VAR_NORM_RT_SCORE',
      'VAR_XCORR_COELUTION',
      'VAR_XCORR_COELUTION_WEIGHTED',
      'VAR_XCORR_SHAPE',
      'VAR_XCORR_SHAPE_WEIGHTED',
      'VAR_YSERIES_SCORE',
    ]),
    'SCORE_MS2': list([
      'SCORE',
      'PVALUE',
      'PEP',
      'QVALUE',
    ]),
    'SCORE_PEPTIDE': list([
      'SCORE',
      'PVALUE',
      'PEP',
      'QVALUE',
    ]),
    'SCORE_PROTEIN': list([
      'SCORE',
      'PVALUE',
      'PEP',
      'QVALUE',
    ]),
  })
# ---
